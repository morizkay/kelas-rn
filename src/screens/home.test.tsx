import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import renderer from "react-test-renderer";
import HomeScreen from "./Home";

describe("Test Home Screen", () => {
  test("Snapshoot Home Screen", () => {
    const tree = renderer
      .create(<HomeScreen navigation={undefined} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Test TouchableOpacity Home Screen", () => {
    const mockNavigation = {
      navigate: (value) => value,
    };
    const { getByTestId } = render(<HomeScreen navigation={mockNavigation} />);
    const getButton = getByTestId("pressable-logistic");
    const handleButton = fireEvent.press(getButton);
    expect(handleButton).toBe("LogisticHome");
  });
});
