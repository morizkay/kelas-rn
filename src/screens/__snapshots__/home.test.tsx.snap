// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Screen Snapshoot Home Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00a680",
        "borderRadius": 10,
        "marginBottom": 10,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "textTransform": "capitalize",
        }
      }
    >
      Logistic's Data
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00a680",
        "borderRadius": 10,
        "marginBottom": 10,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "textTransform": "capitalize",
        }
      }
    >
      Movie's Data
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00a680",
        "borderRadius": 10,
        "marginBottom": 10,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "textTransform": "capitalize",
        }
      }
    >
      Pokemon's Data
    </Text>
  </View>
</View>
`;
